icu4c/source/common/dictbe.cpp

```diff
@@ -1385,15 +1419,54 @@ CjkBreakEngine::divideUpDictionaryRange( UText *inText,
     }
     (void)prevCPPos; // suppress compiler warnings about unused variable
 
+    UChar32 nextChar = utext_char32At(inText, rangeEnd);
     if (!foundBreaks.isEmpty() && foundBreaks.peeki() == rangeEnd) {
-        foundBreaks.popi();
-        correctedNumBreaks--;
+        // In phrase breaking, there has to be a breakpoint between Cj character and
+        // the number/open punctuation.
+        // E.g. る文字「そうだ、京都」->る▁文字▁「そうだ、▁京都」-> breakpoint between 字 and「
+        // E.g. 乗車率９０％程度だろうか -> 乗車▁率▁９０％▁程度だ▁ろうか -> breakpoint between 率 and ９
+        if (isPhraseBreaking) {
+            if (!fOpenPunctuationSet.contains(nextChar) && !fNumberWordSet.contains(nextChar)) {
```

Comment:
merge fOpenPunctuationSet and fNumberWordSet to fNumberOrOpenPunctuationSet
so you only need to have one test
```
if (!fNumberOrOpenPunctuationSet.contains(nextChar)) {
```
here

Reply:
Resolved